cpp/models/abm/lockdown_rules.cpp

```diff
@@ -30,18 +31,21 @@ void set_home_office(TimePoint t_begin, double p, MigrationParameters& params)
 {
     auto damping1 = Eigen::VectorXd::Constant(1, p);
     params.get<WorkRatio>().add_damping(damping1, SimulationTime(t_begin.days()));
+    params.get<WorkRatio>().finalize();
```

Comment:
maybe we can gather all the finalize calls somewhere to make it clear when this should be called (outside of parallel regions).

But we can rethink the finalize in general. Quick examplanation of finalize: Dampings do some computation (adding up all the dampings) and cache the result. The cache is created on demand (e.g., by calling `get_matrix_at`, not thread safe) or by calling finalize. So we call finalize after adding dampings to prepare the cache before entering a parallel region, then `get_matrix_at` just reads the existing cache, which is thread safe. 

I don't think giving every damping a mutex to snychronize the cache so finalize is a good idea, mutexes are heavy and difficult to move around, dampings are a basic building block and should be moveable/copieable. But we might make finalize unnecessary in most cases by switching the defaults. In most cases (like in the example here), immediately recomputing the cache after adding dampings is fine. But in some cases this is expensive (e.g., dynamic NPIs, which could add quite a lot of dampings at many time steps during the simulation). We could make it so the cache needs to be explicitly disabled (and enabled after adding dampings), otherwise the cache is immediatly recomputed on every `add_damping`. Then the default is for safety. Right now the default is for performance.

Reply:
Maybe a new issue?

Reply:
#753